docs/DEVICE-CONNECTIVITY.md

```diff
@@ -1,142 +1,483 @@
+to ensure connectivity with the cloud.
+
+First of all, before configuring a device, the user must find out if there is a network proxy
+in the network and understand how it is configured. Configuration will then depend on the type
+of the proxy, see below.
+
+### Transparent network proxy
+
+A network proxy may operate *transparently*, meaning that clients (EVE devices, user apps)
+making HTTP/HTTPS/… requests are not aware of its presence. When a client makes an HTTP(S)
+request, it addresses it directly to the destination (e.g. EVE controller), but a transparent
+proxy will capture it (without the client knowing about it) and apply a proxying action.
+The action may depend on the destination URL. For example, the proxy may be configured to drop
+requests heading to some specific domains, but allow others.
+
+Additionally, the proxy may break the TLS tunnel into two (client-proxy, proxy-destination)
+and act as a MITM, reading the otherwise encrypted traffic. This is done by proxy generating
+server certificate for the destination on the fly and presenting it to the client during
+the TLS handshake. Normally, the client would detect this because it would not trust such
+a certificate, after all it is signed by proxy CA which is not in the default set of trusted
+certificates installed in EVE/app. However, if MITM proxying is desired (e.g. when company policy
+requires that all traffic, including encrypted, is inspected), then the device has to be told
+to trust the proxy CA certificate. This is why EVE allows the user to insert a proxy CA
+certificate as part of the transparent proxy config. Since everything else is transparent,
+there are no additional config options.
+
+Please note that EVE logs info about the certificate presented by "the other side",
+which in case of MITM proxy would be the proxy and not the controller:
+
+```go
+// In client.go:
+log.Noticef("Peer certificate:(%d) Issuer: %s, Subject: %s, NotAfter: %v",
+    i, cert.Issuer, cert.Subject, cert.NotAfter)
+```
 
-This is accomplished by logic to test connectivity to the controller (implemented in the [DPCManager](../pkg/pillar/dpcmanager), which is part of the Network Interface Manager [nim](../pkg/pillar/cmd/nim)), and maintaining a list of current working and fallback configuration in ```/persist/status/nim/DevicePortConfigList/global.json```
+Information about TLS tunnels opened and peer certificates presented is also part
+of the [network tracing](#netdump-and-nettrace).
 
-To handle different types of connectivity towards the controller EVE supports both load spreading and failover when it comes to maintaining connectivity to the controller.
+### Explicit network proxy
 
-## Load spreading and failover
+Presence of *explicit* proxy is known to the client making HTTP/HTTPS/… requests.
+The client will ask the proxy explicitly to proxy a given request - e.g. make connection
+for me to the controller and perform a GET request for path xyz, then return the result
```

Comment:
I don't know how much detail you want to get into here, but it could be a POST or a GET. An "URL" might be a better term than "path" here.

But less detail would be e.g., "will explicitly ask the proxy to make a HTTPS request to a specified URL."

Reply:
Updated the sentence as suggested.